cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/DefaultStore.java

```diff
@@ -559,6 +559,29 @@ public void scanApplications(int txBatchSize, BiConsumer<ApplicationId, Applicat
     }
   }
 
+  @Override
+  public void scanApplications(NamespaceId id, int txBatchSize,
```

Comment:
This method is similar to the scanApplications which does not take in NamespaceId as a parameter. I considered refactoring the two to avoid duplicate code, but given the lambda seemed sufficiently different i did not refactor it.

However, if the reviewers deem it appropriate, I can refactor it.